siddhi-testsuite/src/main/resources/archetype-resources/component/src/test/java/sample/AbstractTemperatureAlertTests.java

```diff
@@ -0,0 +1,152 @@
+package sample;
+
+import io.siddhi.core.SiddhiAppRuntime;
+import io.siddhi.core.SiddhiManager;
+import io.siddhi.core.stream.input.InputHandler;
+import io.siddhi.distribution.test.framework.MySQLContainer;
+import io.siddhi.distribution.test.framework.NatsContainer;
+import io.siddhi.distribution.test.framework.SiddhiRunnerContainer;
+import io.siddhi.distribution.test.framework.util.NatsClient;
+import io.siddhi.extension.io.nats.sink.NATSSink;
+import io.siddhi.extension.map.json.sinkmapper.JsonSinkMapper;
+import org.testcontainers.containers.output.WaitingConsumer;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Class for AbstractTemperatureAlertTests of Temp-Alert-App. This class includes the common logic for integration
+ * tests and blackbox tests.
+ *
+ * Description: Used for temperature monitoring and anomaly detection. Consumes events from a Nats topic,
+ *              filters the event under types 'monitored' and 'internal'.
+ *              Monitored events are then sent through a pattern and the matched events will be alerted to a Nats topic.
+ *              The internal events are persisted to a table.
+ */
+public abstract class AbstractTemperatureAlertTests {
+
+    public WaitingConsumer siddhiLogConsumer = new WaitingConsumer();
+
+    public MySQLContainer mySQLContainer;
+    public NatsContainer natsContainer;
+    public SiddhiRunnerContainer siddhiRunnerContainer;
+    public NatsClient natsClient;
+    private String natsClusterId;
+    private String natsUrl;
+    private String natsInputDestination;
+    private String natsOutputDestination;
+
+    @BeforeClass
+    public abstract void setUpCluster() throws IOException, InterruptedException;
+
+    @AfterClass
+    public abstract void shutdownCluster();
+
+    public void configureNatsConnection(String natsClusterId, String natsUrl, String natsInputDestination,
+                                        String natsOutputDestination) {
+        this.natsClusterId = natsClusterId;
+        this.natsUrl = natsUrl;
+        this.natsInputDestination = natsInputDestination;
+        this.natsOutputDestination = natsOutputDestination;
+    }
+
+    @Test
+    public void testMessageConsumption() throws InterruptedException, IOException, TimeoutException {
+        natsClient.publish(natsInputDestination, "{\n" +
+                "    \"event\": {\n" +
+                "        \"type\": \"dummyType\",\n" +
+                "        \"deviceID\": \"dummyDeviceID\",\n" +
+                "        \"temp\": 50.0,\n" +
+                "        \"roomID\": \"dummyRoomID\"\n" +
+                "    }\n" +
+                "}");
+        try {
+            siddhiLogConsumer.waitUntil(frame ->
```

Comment:
You cannot assert from logConsumer for Blackbox teste right? Update the Temperature app to publish the result to the log service and assert from there